src/main/java/gregtech/api/recipes/Recipe.java

```diff
@@ -84,21 +84,15 @@ public Recipe(List<GTRecipeInput> inputs, List<ItemStack> outputs, List<ChanceEn
                   IRecipePropertyStorage recipePropertyStorage) {
         this.recipePropertyStorage =
                 recipePropertyStorage == null ? EmptyRecipePropertyStorage.INSTANCE : recipePropertyStorage;
-        if (inputs.isEmpty()) {
-            this.inputs = Collections.emptyList();
-        } else {
-            this.inputs = NonNullList.create();
-            this.inputs.addAll(inputs);
-            this.inputs.sort((ing1, ing2) -> Boolean.compare(ing1.isNonConsumable(), ing2.isNonConsumable()));
```

Comment:
this sorting of noncosumables will need to be moved to the GTRecipeWrapper so JEI still shows them first, for aesthetically purposes

Reply:
For now the sort is performed inside `GTRecipeInputCache#deduplicateInputs`. Should this be removed and added to `GTRecipeWrapper` instead?

Although kind of unrelated here, in my addon I wrote some [cursed shit](https://gist.github.com/Tictim/8734de9f4ed9bc23a7b24dab221d92c7) because I wanted to put certain nonconsumable ingredient on first slot instead of last. I just lived with it because the situation were kind of addon-specific, but should a more extensive solution be devised?